src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -45,6 +48,16 @@ public CommandResult execute(String commandText) throws CommandException, ParseE
         Command command = addressBookParser.parseCommand(commandText);
         commandResult = command.execute(model);
 
+        /* DISPLAY ANY OTHER UIs */
+        //ViewClientCommand has an additional UI to display profile
+        if (command instanceof ViewClientCommand) {
+            ViewClientCommand viewClient = (ViewClientCommand) command;
+            Stage clientView = new Stage();
+            ClientInfoPage cip = new ClientInfoPage(viewClient.getClient());
+            cip.start(clientView);
+            cip.show();
+        }
+
```

Comment:
Perhaps its better to move this to `ViewClientCommand::execute`?
as to maintain separation between `Ui` and `Logic`  components?

Reply:
Yes. I agree with you, but UI cannot run test cases. 
Putting it in ViewClientCommand cannot test execute. 

Usually, it should be placed in MainWindow::executeCommand, but I will see how

Reply:
Sounds good!